src/python/TaskWorker/Actions/RucioDataDiscovery.py

```diff
@@ -0,0 +1,467 @@
+            else:
+                tapeRecallStatus = 'SUBMITFAILED'
+            configreq = {'workflow': self.taskName,
+                         'taskstatus': tapeRecallStatus,
+                         'ddmreqid': ruleId,
+                         'subresource': 'addddmreqid',
+                         }
+            try:
+                tapeRecallStatusSet = self.crabserver.post(api='task', data=urlencode(configreq))
+            except HTTPException as hte:
+                self.logger.exception(hte)
+                msg = "HTTP Error while contacting the REST Interface %s:\n%s" % (
+                    self.config.TaskWorker.restHost, str(hte))
+                msg += "\nStoring of %s status and ruleId (%s) failed for task %s" % (
+                    tapeRecallStatus, ruleId, self.taskName)
+                msg += "\nHTTP Headers are: %s" % hte.headers
+                raise TaskWorkerException(msg, retry=True)
+            if tapeRecallStatusSet[2] == "OK":
+                self.logger.info("Status for task %s set to '%s'", self.taskName, tapeRecallStatus)
+            if automaticTapeRecallIsImplemented:
+                msg += "\nThis task will be automatically submitted as soon as the stage-out is completed."
+                self.uploadWarning(msg, self.userproxy, self.taskName)
+                raise TapeDatasetException(msg)
+            # fall here if could not setup for automatic submission after recall
+            msg += "\nPlease monitor recall progress via Rucio or DAS and try again once data are on disk."
+            raise TaskWorkerException(msg)
+
+        if system == 'None':
+            msg += '\nIt is not possible to request a recall from tape.'
+            msg += "\nPlease, check DAS (https://cmsweb.cern.ch/das) and make sure the dataset is accessible on DISK."
+            raise TaskWorkerException(msg)
+
+        if system == 'Dynamo':
+            raise NotImplementedError
+
+    def execute(self, *args, **kwargs):
+        """
+        This is a convenience wrapper around the executeInternal function
+        """
+
+        # DBS3 requires X509_USER_CERT to be set - but we don't want to leak that to other modules
+        # so use a context manager to set an ad hoc env and restore as soon as
+        # executeInternal is over, even if it raises exception
+
+        with self.config.TaskWorker.envForCMSWEB:
+            result = self.executeInternal(*args, **kwargs)
+
+        return result
+
+    def executeInternal(self, *args, **kwargs):
+
+        self.logger.info("Data discovery with Rucio") ## to be changed into debug
+
+        self.taskName = kwargs['task']['tm_taskname']           # pylint: disable=W0201
+        self.username = kwargs['task']['tm_username']           # pylint: disable=W0201
+        self.userproxy = kwargs['task']['user_proxy']           # pylint: disable=W0201
+
+        splitting = kwargs['task']['tm_split_algo']
+        if splitting != 'FileBased':
```

Comment:
Should we reject it at the REST?

Reply:
I fear that if we do in the RESY it is difficult to give back a clear error mesage, but I have not looeked into it. I guess I was simply lazily following  what happens with other "incompatibilities' e.g. in UserDataDiscovery. Yes, worth looking at.

Reply:
a "second order" correction. I will track it in #7538 which will be kept open